internal/security/anchore.go

```diff
@@ -0,0 +1,237 @@
+	config := database.Config{
+		Host:      viper.GetString("anchore.host"),
+		Port:      viper.GetInt("anchore.port"),
+		User:      viper.GetString("anchore.user"),
+		Pass:      viper.GetString("anchore.password"),
+		Name:      viper.GetString("anchore.dbname"),
+		EnableLog: viper.GetBool("anchore.logging"),
+	}
+
+	err := config.Validate()
+	if err != nil {
+		logger.Panic("invalid Anchore database config: ", err.Error())
+	}
+
+	dbConnStr := fmt.Sprintf("host=%v port=%v user=%v dbname=%v password=%v sslmode=disable", config.Host, config.Port, config.User, config.Name, config.Pass)
+	logrus.Info(dbConnStr)
+	db, err := gorm.Open("postgres", dbConnStr)
+	if err != nil {
+		logger.Panic("failed to initialize db: ", err.Error())
+	}
+	return &AnchoreDB{database: db}
+}
+
+type User struct {
+	UserId         string `gorm:"primary_key;column:userId"`
+	CreatedAt      int64
+	LastUpdated    int64
+	RecordStateKey string
+	RecordStateVal string
+	Password       string
+	Email          string
+	Acls           string
+	Active         bool
+}
+
+func (db *AnchoreDB) findAnchoreUser(name string) (*User, error) {
+	user := User{UserId: name}
+	err := db.database.Where(&user).First(&user).Error
+	if err != nil {
+		return nil, err
+	}
+	return &user, nil
+}
+
+func (db *AnchoreDB) createAnchoreUser(name string, password string) error {
+	user := User{
+		UserId:         name,
+		Password:       password,
+		CreatedAt:      time.Now().Unix(),
+		LastUpdated:    time.Now().Unix(),
+		Email:          "banzai@banzaicloud.com",
+		Active:         true,
+		RecordStateKey: "active",
+	}
+	err := db.database.Create(&user).Error
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//DeleteAnchoreUser
+func (db *AnchoreDB) DeleteAnchoreUser(name string) error {
+	user := User{
+		UserId: name,
+	}
+	err := db.database.Delete(&user).Error
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+//SetupAnchoreUser sets up a new user in Anchore Postgres DB & creates / updates a secret containng user name /password.
+func SetupAnchoreUser(orgId uint, clusterId string) (bool, error) {
+	if !AnchorEnabled {
+		logger.Infof("Anchore integration is not enabled.")
+		return false, nil
+	}
+	anchorUserName := fmt.Sprintf("%v-anchore-user", clusterId)
+	db := NewAnchoreDB()
+	defer db.database.Close()
+
+	user, err := db.findAnchoreUser(anchorUserName)
+	var userPassword string
+	if err != nil {
+		logger.Infof("Anchore user %v not found, creating", anchorUserName)
+		userPassword, err = password.Generate(16, 4, 4, false, true)
+		if err != nil {
+			return true, errors.WrapPrefix(err, "Error generating password for Anchor user", 0)
+		}
+		err := db.createAnchoreUser(anchorUserName, userPassword)
+		if err != nil {
+			return true, errors.WrapPrefix(err, "Error creating Anchor user", 0)
+		}
+	} else {
+		logger.Infof("Anchore user %v found", anchorUserName)
+		userPassword = user.Password
+	}
+
+	secretRequest := secret.CreateSecretRequest{
```

Comment:
If you submit a CreateSecretRequest with type password, with an empty password field, the secret engine will generate a password for you.